pkg/reconciler/buildrun/resources/errors.go

```diff
@@ -9,14 +9,52 @@ import (
 	"strings"
 )
 
-// HandleError returns multiple errors if each error is not nil.
-// And its error message.
-func HandleError(message string, listOfErrors ...error) error {
-	var errSlice []string
-	for _, e := range listOfErrors {
-		if e != nil {
-			errSlice = append(errSlice, e.Error())
+// ClientStatusUpdateError is an error that occurrs when trying to update a runtime object status
+type ClientStatusUpdateError struct {
+	reason error
+}
+
+func (e ClientStatusUpdateError) Error() string {
+	return fmt.Sprintf("failed to update client status: %v", e.reason)
+}
+
+// IsClientStatusUpdateError checks whether the given error is of type ClientStatusUpdateError
```

Comment:
```suggestion
// IsClientStatusUpdateError checks whether the given error is of type ClientStatusUpdateError or
// in case this is a list of errors, that it contains at least one error of type ClientStatusUpdateError
```

Reply:
done